src/transformers/models/ict/configuration_ict.py

```diff
@@ -0,0 +1,299 @@
+
+
+logger = logging.get_logger(__name__)
+
+ICT_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "sheonhan/ict-imagenet-32": "https://huggingface.co/sheonhan/ict-imagenet-32/resolve/main/config.json",
+}
+
+
+class ICTGuidedUpsamplerConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`ICTGuidedUpsampler`]. It is used to instantiate an
+    [`ICTGuidedUpsampler`] model according to the specified arguments, defining the model architecture. Instantiating a
+    configuration with the defaults will yield a similar configuration to that of the [ICTGuidedUpsampler model trained with the ImageNet dataset](https://huggingface.co/sheonhan/ict-imagenet-32).
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+
+    Args:
+        learning_rate (`float`, *optional*, defaults to 0.0001):
+            The desired learning rate of the [`ICTGuidedUpsampler`] model.
+        dis_gen_learning_rate (`float`, *optional*, defaults to 0.1):
+            The discriminator/generator learning rate ratio.
+        adam_beta1 (`float`, *optional*, defaults to 0.0):
+            The beta1 to use in Adam.
+        adam_beta2 (`float`, *optional*, defaults to 0.9):
+            The beta2 to use in Adam.
+        batch_size (`int`, *optional*, defaults to 64):
+            The batch size for training.
+        input_size (`int`, *optional*, defaults to 256):
+            The input image size for training. (0 for the original size.)
+        max_iteration (`float`, *optional*, defaults to 5e7):
+            The maximum number of iterations to train the model.
+        residual_blocks (`int`, *optional*, defaults to 8):
+            The number of residual blocks.
+        l1_loss_weight (`float`, *optional*, defaults to 1.0):
+            The weight of the L1 loss function.
+        style_loss_weight (`float`, *optional*, defaults to 250.):
+            The weight of the style loss module.
+        content_loss_weight (`float`, *optional*, defaults to 0.1):
+            The weight of the content loss module.
+        inpaint_adv_loss_weight (`float`, *optional*, defaults to 0.1):
+            The weight of the adversial loss module.
+        gan_loss (`str`, *optional*, defaults to`"nsgan"`):
+            GAN's loss function can be either "nsgan", "lsgan", or "hinge".
+
+
+    Example:
+
+    ```python
+    >>> from transformers import ICTGuidedUpsamplerConfig, ICTGuidedUpsampler
+
+    >>> # Initializing a ICT ict-imagenet-32 style configuration
+    >>> configuration = ICTGuidedUpsamplerConfig()
+
+    >>> # Initializing a model (with random weights) from the ict-imagenet-32 style configuration
+    >>> upsampler = ICTGuidedUpsampler(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = upsampler.config
+    ```"""
+    model_type = "ict-guided-upsampler"
```

Comment:
Not sure if this value is correct

Reply:
This would be better:
` model_type = "ict_guided_upsampler"`